### PR TITLE
[Metricbeat] Add latency config option into aws module

### PR DESCRIPTION
**Pull request by kaiyan-sheng**
## What does this PR do?

This PR adds the `latency` config parameter for `aws` module. With this config parameter, users can collect CloudWatch metrics with a latency larger than Metricbeat collection period. For example, for S3 request metrics, the latency can be 3 minutes. The recommended collection period is 1min. Without the `latency` config parameter, no data points will be collected ever. 

In this screenshot: current timestamp is 14:21 but we only see the latest data at 14:18 with a 3-minute delay. In this case, without this fix, our Metricbeat module will keep trying to collect data from the last 2-minute time frame, which will always be empty:
Screenshot 2020-09-03 at 16 22 19

(@gbanasiak Thank you for your screenshot!)

## Why is it important?

`latency` config sets CloudWatch API start time and end time with a delay. For example, if the current time is `2020-08-31T18:30:00.000Z` with collection period `1m`. 

Without `latency`:
startTime = `2020-08-31T18:29:00.000Z`
endTime = `2020-08-31T18:30:00.000Z`

With `latency` set to `5min`:
startTime = `2020-08-31T18:24:00.000Z`
endTime = `2020-08-31T18:25:00.000Z`

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally
1. Enable aws module: `./metricbeat modules enable aws`
2. Change `modules.d/aws.yml` to:
```
- module: aws
  period: 5m
  credential_profile_name: elastic-beats
  latency: 5m
  metricsets:
    - ec2
```
3. Start Metricbeat with debug level logging: `./metricbeat -e -d "*"`
4. You should see the startTime and endTime in debug log with a latency of 5 minutes applied.

### CODE REVIEW

File: metricbeat/docs/modules/aws.asciidoc [hunk 1]
```diff
@@ -19,16 +19,27 @@ module. Please see <<aws-api-requests,AWS API requests>> for more details.
...
+* *regions*
+
 This module also accepts optional configuration `regions` to specify which
 AWS regions to query metrics from. If the `regions` parameter is not set in the
 config file, then by default, the `aws` module will query metrics from all available
 AWS regions.
 
+* *latency*
+
+Some AWS services send monitoring metrics to CloudWatch with a latency to
```
Comment (ChrsMark) [inline metricbeat/docs/modules/aws.asciidoc:38]: Do we already know these AWS services? If so, maybe to list them here to make it easy for new users?

**Review by ChrsMark — COMMENTED** [review]
lgtm! just one comment regarding docs.

File: metricbeat/docs/modules/aws.asciidoc (hunk 1 above)
Comment (gbanasiak) [reply metricbeat/docs/modules/aws.asciidoc:38]: So far we have identified one - S3.

Comment (kaiyan-sheng) [reply metricbeat/docs/modules/aws.asciidoc:38]: Yeah right now it's only for S3 metrics. But with `cloudwatch` as the generic metricset, I assume there will be more in the future that has the latency issue.

**Review by ChrsMark — APPROVED** [review]
lgtm

**Pull request merged**